src/deepsparse/loggers/prometheus_logger.py

```diff
@@ -0,0 +1,116 @@
+    :param text_log_save_dir: the directory where the text log files
+        are saved. By default, the python working directory
+    :param text_log_file_name: the name of the text log file.
+        Default: `prometheus_logs.prom`
+    """
+
+    def __init__(
+        self,
+        port: int = 6100,
+        text_log_save_frequency: int = 10,
+        text_log_save_dir: str = os.getcwd(),
+        text_log_file_name: Optional[str] = None,
+    ):
+
+        self.port = port
+        # until we have done research into persistent
+        # logs, lets save logs as .prom files
+        self.text_log_save_frequency = text_log_save_frequency
+        self.text_log_file_path = os.path.join(
+            text_log_save_dir, text_log_file_name or "prometheus_logs.prom"
+        )
+        self._prometheus_metrics = defaultdict(str)
+
+        self._setup_client()
+        self._counter = 0
+
+    def log(self, identifier: str, value: Any, category: MetricCategories):
+        """
+        Collect information from the pipeline and pipe it them to the stdout
+
+        :param identifier: The name of the thing that is being logged.
+        :param value: The data structure that the logger is logging
+        :param category: The metric category that the log belongs to
+        """
+
+        # needs to adhere to
+        # https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+        formatted_identifier = identifier.replace(".", "__")
+        prometheus_metric = self._prometheus_metrics.get(formatted_identifier)
+        if prometheus_metric is None:
+            prometheus_metric = self._add_metric_to_registry(
+                formatted_identifier, category
+            )
+        prometheus_metric.observe(self._validate(value))
+        self._try_export_metrics_to_textfile()
+
+    def _add_metric_to_registry(self, identifier: str, category: str) -> Summary:
+        description = (
+            """Summary metric for identifier: {identifier} | Category: {category}"""
+        )
+        prometheus_metric = Summary(
+            identifier,
+            description.format(identifier=identifier, category=category),
+            registry=REGISTRY,
+        )
+        self._prometheus_metrics[identifier] = prometheus_metric
+        return prometheus_metric
+
+    def _try_export_metrics_to_textfile(self):
+        if self._counter % self.text_log_save_frequency == 0:
+            write_to_textfile(self.text_log_file_path, REGISTRY)
+            self._counter = 0
+        self._counter += 1
+
+    def _setup_client(self):
+        # starts the Prometheus client
+        start_http_server(port=self.port)
+        _LOGGER.info(f"Prometheus client: started. Using port: {self.port}.")
+
+    def _validate(self, value: Any) -> Any:
+        # make sure we are passing a value that can be
+        # logged by prometheus client
+        if not any(isinstance(value, datatype) for datatype in SUPPORTED_DATA_TYPES):
```

Comment:
ack revising, but fyi 
```suggestion
        if not isinstance(value, SUPPORTED_DATA_TYPES):  # SUPPORTED_DATA_TYPES must be tuple
```